File: setup-vercel-env.py
import requests
import json
import os
import sys

# Get Vercel token from CLI config
vercel_token_path = os.path.expanduser('~/.vercel/auth.json')
try:
    with open(vercel_token_path, 'r') as f:
        auth_data = json.load(f)
        token = auth_data.get('token')
except:
    print("❌ Erro: Não foi possível encontrar o token do Vercel")
    sys.exit(1)

if not token:
    print("❌ Erro: Token do Vercel não encontrado")
    sys.exit(1)

# Environment variables to add
env_vars = {
    'NEXT_PUBLIC_SUPABASE_URL': {
        'value': 'https://wjhsitjnccmfrudukbix.supabase.co',
        'sensitive': False
    },
    'NEXT_PUBLIC_SUPABASE_ANON_KEY': {
        'value': 'eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJzdXBhYmFzZSIsInJlZiI6IndqaHNpdGpuY2NtZnJ1ZHVrYml4Iiwicm9sZSI6ImFub24iLCJpYXQiOjE3NzAxNTc1NTksImV4cCI6MjA4NTczMzU1OX0.WHUchnyNmt3hFdzNdLrQcNUFG-TuqJAO0lvxOiweots',
        'sensitive': False
    },
    'SUPABASE_URL': {
        'value': 'https://wjhsitjnccmfrudukbix.supabase.co',
        'sensitive': False
    },
    'SUPABASE_SERVICE_ROLE_KEY': {
        'value': 'eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJzdXBhYmFzZSIsInJlZiI6IndqaHNpdGpuY2NtZnJ1ZHVrYml4Iiwicm9sZSI6InNlcnZpY2Vfcm9sZSIsImlhdCI6MTc3MDE1NzU1OSwiZXhwIjoyMDg1NzMzNTU5fQ.4o7G3mKad868PjJGWLgC679XmNUcwzjMxxI8lHwnZv0',
        'sensitive': True
    },
    'FRONTEND_ORIGIN': {
        'value': 'https://betarena-kohl.vercel.app',
        'sensitive': False
    },
    'ADMIN_API_KEY': {
        'value': 'betarena-admin-key-2026-super-secret',
        'sensitive': True
    },
    'PORT': {
        'value': '3001',
        'sensitive': False
    }
}

headers = {
    'Authorization': f'Bearer {token}',
    'Content-Type': 'application/json'
}

project_id = 'betarena'
team_id = 'joao-adrianos-projects-024efa77'

print("🔐 Adicionando variáveis de ambiente ao Vercel...")
print("=" * 80)

for var_name, var_data in env_vars.items():
    url = f'https://api.vercel.com/v9/projects/{project_id}/env'
    
    payload = {
        'key': var_name,
        'value': var_data['value'],
        'target': ['production', 'preview', 'development'],
        'type': 'sensitive' if var_data['sensitive'] else 'plain'
    }
    
    try:
        response = requests.post(url, headers=headers, json=payload, params={'teamId': team_id})
        
        if response.status_code in [200, 201]:
            print(f"✅ {var_name}: Adicionado com sucesso")
        else:
            print(f"⚠️  {var_name}: Erro {response.status_code}")
            print(f"   Resposta: {response.text}")
    except Exception as e:
        print(f"❌ {var_name}: Erro na requisição - {str(e)}")

print("=" * 80)
print("✨ Variáveis de ambiente adicionadas!")
print("\n📝 Próximo passo: Faça redeploy no Vercel")
print("   npx vercel --prod")

File: setup-vercel-env.js
#!/usr/bin/env node
/**
 * Script para adicionar variáveis de ambiente ao Vercel via curl/fetch
 * Executar: node setup-vercel-env.js
 */

const fs = require('fs');
const path = require('path');
const { execSync } = require('child_process');

// Read token from Vercel CLI config
const tokenPath = path.join(process.env.HOME || process.env.USERPROFILE, '.vercel', 'auth.json');
let token = '';

try {
  const authData = JSON.parse(fs.readFileSync(tokenPath, 'utf-8'));
  token = authData.token;
} catch (error) {
  console.error('❌ Erro: Não foi possível encontrar o token do Vercel em:', tokenPath);
  console.error('   Faça login com: npx vercel login');
  process.exit(1);
}

const projectId = 'betarena';
const teamId = 'joao-adrianos-projects-024efa77';

const envVars = [
  { key: 'NEXT_PUBLIC_SUPABASE_URL', value: 'https://wjhsitjnccmfrudukbix.supabase.co', sensitive: false },
  { key: 'NEXT_PUBLIC_SUPABASE_ANON_KEY', value: 'eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJzdXBhYmFzZSIsInJlZiI6IndqaHNpdGpuY2NtZnJ1ZHVrYml4Iiwicm9sZSI6ImFub24iLCJpYXQiOjE3NzAxNTc1NTksImV4cCI6MjA4NTczMzU1OX0.WHUchnyNmt3hFdzNdLrQcNUFG-TuqJAO0lvxOiweots', sensitive: false },
  { key: 'SUPABASE_URL', value: 'https://wjhsitjnccmfrudukbix.supabase.co', sensitive: false },
  { key: 'SUPABASE_SERVICE_ROLE_KEY', value: 'eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJzdXBhYmFzZSIsInJlZiI6IndqaHNpdGpuY2NtZnJ1ZHVrYml4Iiwicm9sZSI6InNlcnZpY2Vfcm9sZSIsImlhdCI6MTc3MDE1NzU1OSwiZXhwIjoyMDg1NzMzNTU5fQ.4o7G3mKad868PjJGWLgC679XmNUcwzjMxxI8lHwnZv0', sensitive: true },
  { key: 'FRONTEND_ORIGIN', value: 'https://betarena-kohl.vercel.app', sensitive: false },
  { key: 'ADMIN_API_KEY', value: 'betarena-admin-key-2026-super-secret', sensitive: true },
  { key: 'PORT', value: '3001', sensitive: false }
];

async function addEnvVar(variable) {
  const url = `https://api.vercel.com/v9/projects/${projectId}/env?teamId=${teamId}`;
  const payload = {
    key: variable.key,
    value: variable.value,
    target: ['production', 'preview', 'development'],
    type: variable.sensitive ? 'sensitive' : 'plain'
  };

  const curlCmd = `curl -X POST "${url}" -H "Authorization: Bearer ${token}" -H "Content-Type: application/json" -d '${JSON.stringify(payload)}'`;
  
  try {
    const result = execSync(curlCmd, { encoding: 'utf-8' });
    const response = JSON.parse(result);
    
    if (response.id || response.key) {
      console.log(`✅ ${variable.key}: Adicionado com sucesso`);
      return true;
    } else if (response.error) {
      console.log(`⚠️  ${variable.key}: ${response.error.message}`);
      return false;
    }
  } catch (error) {
    console.log(`❌ ${variable.key}: Erro na requisição`);
    return false;
  }
}

async function main() {
  console.log('🔐 Adicionando variáveis de ambiente ao Vercel...');
  console.log('='.repeat(80));

  for (const variable of envVars) {
    await addEnvVar(variable);
  }

  console.log('='.repeat(80));
  console.log('✨ Processo concluído!');
  console.log('\n📝 Próximo passo: Faça redeploy no Vercel');
  console.log('   npx vercel --prod');
}

main();
